### ISSUE TITLE
Institute CLA Policy for Carbon Contributors

### ISSUE THREAD

**Issue by shinytoyrobots** [carbon-design-system/issue-tracking#9]
From @claycrenshaw 

>Some contributors on GH have not filled out their identity information, and are only identifiable by a GH handle. This raises a number of problems, including the inability to put comments, PRs, etc. into a meaningful corporate context. If I need to reach out to someone on Slack, for instance, I have no way of knowing that person's real name. @joshblack

**Comment by shinytoyrobots** [comment]
From @joshblack 

> Example from Facebook: https://github.com/facebook/react/pull/11930#issuecomment-354441132

> cc @kuehndaniel  do we know if this is a requirement for OSS projects for IBM?

**Comment by shinytoyrobots** [comment]
This seems like a very sensible approach.

**Comment by kuehndaniel** [comment]
I've got a message into Legal with this question. Will post an answer when I get it.

**Comment by mattrosno** [comment]
@kuehndaniel did you get an answer from Legal if CLA policy is required?

**Comment by kuehndaniel** [comment]
@mattrosno Apologies, I thought I had passed this along already. We do need contributors to complete a CLA much in the way Facebook is doing. Attached are the templates for individual contributor and corporate contributor. These are super ugly and may drive people away, so we should design a better experience for filling these out (a la Facebook or something else).


[cla-corporate.pdf](https://github.com/carbon-design-system/issue-tracking/files/2738087/cla-corporate.pdf)
[cla-individual.pdf](https://github.com/carbon-design-system/issue-tracking/files/2738088/cla-individual.pdf)

**Comment by joshblack** [comment]
Could definitely do something like: https://code.facebook.com/cla real quickly

**Comment by kuehndaniel** [comment]
Sounds good. We'll want to do a quick pass for design to get it looking more like us first. Adding a design label and putting it on Chris/Anna/Design radar.

**Comment by mattrosno** [comment]
Another example: https://github.com/material-components/material-components/blob/develop/CONTRIBUTING.md#signing-the-cla

**Comment by laurenmrice** [comment]
I made a pass at the designs needed for the CLA screens using similar structure to what fb/material did with the content Daniel provided.

Could we get Clay to do a pass on content for this?

## 1. We need a Carbon Bot with link prompt/guidance content
![1312_example github bot link to cla type](https://user-images.githubusercontent.com/43969356/52733465-69e23180-2f88-11e9-9e12-67084b8ed8d8.png)

## 2. Select CLA type
![1312_select type of cla](https://user-images.githubusercontent.com/43969356/52733559-a7df5580-2f88-11e9-8280-46e5bef8fbcd.png)

## 3. Individual Contributor License Agreement
![1312_individual contributor_no hang_scrollable agreement](https://user-images.githubusercontent.com/43969356/52733575-b463ae00-2f88-11e9-960f-5393ac11b4a1.png)

## 4. Corporate Contributor License Agreement
![1312_corporate contributor_no hang_scrollable agreement](https://user-images.githubusercontent.com/43969356/52733581-b9286200-2f88-11e9-8d29-96c7a18cc7e2.png)

Sketch file- https://ibm.ent.box.com/folder/66831561872

**Comment by mattrosno** [comment]
Placeholder to call out that CLAs are necessary. For specific content to include here, I'd reference how other governance documents call out CLAs.

https://next.carbondesignsystem.com/contributing/governance/#clas

**Comment by joshblack** [comment]
Nice @laurenmrice! 🎉

**Comment by aagonzales** [comment]
This sprint (2/18/19):
- [x] Need content pass
- [ ] Do we have all the necessary requirements?

**Comment by claycrenshaw** [comment]
@mattrosno @kuehndaniel Do we need to do anything to actually _enforce_ the "no contributions without a CLA" policy on the repos? (See point 1 above) Or do we just need to have people's agreement on file? Trying to figure out how this will work operationally. Thanks!

**Comment by claycrenshaw** [comment]
Feedback

Looks good and clean @laurenmrice !

Some specific notes: 

Screen 3
- I don't think we need placeholder text for this kind of form. . . (if anywhere, possibly in the phone number field to show proper formatting)
- For "Public name" put parentheses around `optional`
- For "Public name" we may want to add an info tip to explain what this means, since it's not necessarily obvious.
- The mailing address probably will need more fields (city, state, zip)
- Should probably make telephone number optional if we are allowed to
- The "I agree" checkbox is worded a bit too vaguely. Maybe it's "I agree to the terms and conditions outlined above" or something like that. . . We may need guidance from Legal to get this just right.
- For the sake of clarity, it might make sense to add a scroll handle to the main text box, to indicate that it is scrollable.

Screen 4
- Change "Personal information" to "Corporation information" hey that rhymes!
- Change "Corporate name" to "Corporation name"
- Same note on the mailing address fields
- Same note on optionality of phone number
- For Schedule A, show how the fields would lay out if the user "adds another" (make sure the grouping of name and email are clear)
- For Schedule B, I think we could use some clarifying helper text here too. I have no real idea what it is asking me for...
- Same note on the "I agree" checkbox.

We should also mock up a confirmation screen to show what happens after the user submits the form.

**Comment by kuehndaniel** [comment]
@claycrenshaw obviously we must have the agreed to CLA on file before we can accept external contributions (outside of IBM). I suppose we need a GitHub power user (@mattrosno @joshblack) to weigh in if we can actually have GitHub stop a PR from being submitted without a CLA on file, or if the enforcement will need be a manual check on our end before we accept a PR—or maybe we can have a bot do the check and respond? I bet it's the latter but just want to double check. The former potential option that keeps someone from actually submitting a PR seems like a bad idea and my preference would be the manual/bot check.

**Comment by claycrenshaw** [comment]
Yes, it looks like Google has kind of a hybrid approach, with a bot being verified by a human. Just something to consider if we aren't already working on it.

**Comment by joshblack** [comment]
Ultimately the codeowners on each repo can choose whether or not to merge something in, in this case 👍 We can have a GitHub bot handle verifying PRs from external individuals to check if they have signed already, and notifying them if they have not. Most likely the easiest thing would be a labelling system with "cla: signed" "cla: not signed".

**Comment by claycrenshaw** [comment]
Removing my assignment since I have completed the feedback portion on this issue.

**Comment by mattrosno** [comment]
We still need a CLA.

**Comment by laurenmrice** [comment]
not stale

**Comment by laurenmrice** [comment]
not stale

**Comment by mattrosno** [comment]
not stale

**Comment by connor-leech** [comment]
We're on it!

**Comment by laurenmrice** [comment]
not stale

**Comment by laurenmrice** [comment]
not stale and hopefully done soon

**Comment by laurenmrice** [comment]
lol not stale

**Comment by laurenmrice** [comment]
nope

**Comment by laurenmrice** [comment]
not stale?

**Comment by laurenmrice** [comment]
not stale

**Comment by laurenmrice** [comment]
not stale

**Comment by laurenmrice** [comment]
not stale

**Comment by laurenmrice** [comment]
😭

**Comment by laurenmrice** [comment]
😅💀

**Comment by laurenmrice** [comment]
not staleeeee

**Comment by laurenmrice** [comment]
NOT STALE!!! 😂

**Comment by tw15egan** [comment]
stalebot pls

**Comment by laurenmrice** [comment]
pls keep alive

**Comment by laurenmrice** [comment]
not stale

**Comment by mattrosno** [comment]
Not stale. (Beat you to it @laurenmrice!)
